src/amo/components/AddAddonToCollection/index.js

```diff
@@ -0,0 +1,301 @@
+    // not server side.
+    this.loadDataIfNeeded();
+  }
+
+  componentWillReceiveProps(nextProps: Props) {
+    this.loadDataIfNeeded(nextProps);
+  }
+
+  loadDataIfNeeded(nextProps?: Props) {
+    const allProps = { ...this.props, ...nextProps };
+    const {
+      dispatch,
+      errorHandler,
+      loadingUserCollections,
+      userCollections,
+      siteUserId,
+    } = allProps;
+
+    if (errorHandler.hasError()) {
+      // Abort loadng data so that the error can be rendered.
+      return;
+    }
+
+    if (siteUserId && !loadingUserCollections && !userCollections) {
+      dispatch(fetchUserCollections({
+        errorHandlerId: errorHandler.id, userId: siteUserId,
+      }));
+    }
+  }
+
+  onSelectOption = (event: ElementEvent<HTMLSelectElement>) => {
+    event.preventDefault();
+    const key = event.target.value;
+    const handleOption = this.optionSelectHandlers[key];
+    if (handleOption) {
+      handleOption();
+    } else {
+      log.warn(`No handler for option: "${key}"`);
+    }
+  }
+
+  addToCollection(collection: CollectionType) {
+    const { addon, errorHandler, dispatch, siteUserId } = this.props;
+    if (!addon) {
+      throw new Error(
+        'Cannot add to collection because no add-on has been loaded');
+    }
+    if (!siteUserId) {
+      throw new Error(
+        'Cannot add to collection because you are not signed in');
+    }
+
+    dispatch(addAddonToCollection({
+      addonId: addon.id,
+      collectionId: collection.id,
+      collectionSlug: collection.slug,
+      errorHandlerId: errorHandler.id,
+      userId: siteUserId,
+    }));
+  }
+
+  createOption(
+    {
+      text, key, onSelect,
+    }: {
+      // eslint-disable-next-line react/no-unused-prop-types
+      text: string, key: string, onSelect?: OnSelectOptionType,
+    }
+  ) {
+    if (onSelect) {
+      this.optionSelectHandlers[key] = onSelect;
+    }
+    return (
+      <option
+        className="AddAddonToCollection-option"
+        key={key}
+        value={key}
+      >
+        {text}
+      </option>
+    );
+  }
+
+  getSelectData(): SelectData {
+    const {
+      _window,
+      i18n,
+      loadingAddonsInCollections,
+      loadingUserCollections,
+      userCollections,
+    } = this.props;
+
+    let progressMessage;
+    if (loadingUserCollections) {
+      progressMessage = i18n.gettext('Loading...');
+    } else if (loadingAddonsInCollections) {
+      progressMessage = i18n.gettext('Adding...');
+    }
+    if (progressMessage) {
+      // Create a disabled select box with a single option.
+      return {
+        disabled: true,
+        options: [
+          this.createOption({
+            text: progressMessage, key: 'default',
+          }),
+        ],
+      };
+    }
+
+    const options = [
+      this.createOption({
+        text: i18n.gettext('Add to collection'), key: 'default',
+      }),
+    ];
+
+    options.push(this.createOption({
+      text: i18n.gettext('Create new collection'),
```

Comment:
I wonder if a little emoji or glyph or something would help this stand out? Or maybe a blank line underneath it (eg `-----`)? I just feel like using this, it's not clear this is a separate action compared to adding to an existing collection, which I think is important to highlight.

Reply:
I put it in an optgroup but we can do some follow-ups if needed.


<img width="628" alt="screenshot 2018-01-17 15 15 11" src="https://user-images.githubusercontent.com/55398/35067616-1da24a6c-fb9a-11e7-9fd3-f7a781cf2f8d.png">

Reply:
Screenshot: https://user-images.githubusercontent.com/55398/35067665-45066d5e-fb9a-11e7-8d00-06f261e35880.png

Reply:
Oh, that's totally nice, thanks 👍